src/Jackalope/Transport/DoctrineDBAL/Query/QOMWalker.php

```diff
@@ -690,12 +690,7 @@ public function walkNumComparisonConstraint(QOM\PropertyValueInterface $property
             );
         }
 
-        return sprintf(
-            '%s %s %s',
-            $this->sqlXpathExtractNumValue($alias, $property),
-            $operator,
-            $literalOperand->getLiteralValue()
-        );
+        return $this->sqlXpathComparePropertyValue($alias, $property, $literalOperand->getLiteralValue(), $operator);
```

Comment:
I actually wonder if we shouldn't do this for all implemenations anyway - as that method handles mysql, postgres and sqlite. The logic for MySQL is however different.